lib/Conversion/TorchToLinalg/Reduction.cpp

```diff
@@ -400,8 +400,17 @@ class ConvertReductionOp : public ConversionPattern {
     bool err = false;
     auto reductionBodyBuilder = [&](OpBuilder &builder, Location loc,
                                     ValueRange payloadArgs) {
-      Value result = createLinalgPayloadForReduceOp(builder, loc, payloadArgs,
-                                                    op, operands, elemType);
+      Value result;
+      // Check if the element type is bool.
+      if (elemType.isInteger(1)) {
+        result = createLinalgPayloadForReduceOp(
+            builder, loc, payloadArgs, op, operands,
+            mlir::IntegerType::get(op->getContext(), 64));
+        result = convertScalarToDtype(rewriter, loc, result, elemType);
+      } else {
+        result = createLinalgPayloadForReduceOp(builder, loc, payloadArgs, op,
+                                                operands, elemType);
+      }
```

Comment:
I don't think this file needs any modifications. It seems to me that the issue is with `RefineTypes`:

https://github.com/llvm/torch-mlir/blob/2bb5f4d8fe1f91715716f46b9b8b895aec078021/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L932

The result type should be `int64` whenever the input is a bool tensor.

Reply:
Thanks, Ramiro for your valuable suggestions.